pkg/pprof/cell_test.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package pprof
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"go.uber.org/goleak"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+func TestPprofHandlers(t *testing.T) {
+	hive := hive.New(
+		Cell,
+		cell.Config(Config{
+			Pprof:     true,
+			PprofPort: 8080,
```

Comment:
This is a really often used port so this is likely to generate a lot of flakes. Preferably we'd allow the kernel to allocate the port, e.g. by using a custom `net.Listener` and then querying what it got via `Addr()`. Or at the very least let's pick a more uncommon port.